src/LinearMaps.jl

```diff
@@ -57,9 +64,9 @@ function LinearAlgebra.mul!(Y::AbstractMatrix, A::LinearMap{T}, X::AbstractMatri
     return Y
 end
 
-A_mul_B!(y::AbstractVector, A::AbstractMatrix, x::AbstractVector)  = mul!(y, A, x)
-At_mul_B!(y::AbstractVector, A::AbstractMatrix, x::AbstractVector) = mul!(y, transpose(A), x)
-Ac_mul_B!(y::AbstractVector, A::AbstractMatrix, x::AbstractVector) = mul!(y, adjoint(A), x)
+@inline A_mul_B!(y::AbstractVector, A::AbstractMatrix, x::AbstractVector)  = mul!(y, A, x)
+@inline At_mul_B!(y::AbstractVector, A::AbstractMatrix, x::AbstractVector) = mul!(y, transpose(A), x)
+@inline Ac_mul_B!(y::AbstractVector, A::AbstractMatrix, x::AbstractVector) = mul!(y, adjoint(A), x)
```

Comment:
This too, I think, is not at all necessary. Such one-line definitions are automatically inlined by the compiler.

Reply:
Hm, pretty much all the one-line `mul!`s have an `@inline` macro in `LinearAlgebra`. When I was experimenting with the 5-arg mul!, I thought I had to add the macro somewhere to avoid a single allocation. But in that other PR, we have tests for that, so the situation might have been different. I think you're right that we shouldn't overload the code unless necessary for performance.

Reply:
That's strange, i.e. the comment about the `LinearAlgebra` statements. Well, I guess they do no harm so you can certainly leave them be.

Reply:
Yes, I just benchmarked `mul!(y, LinearMap(A), x)` vs. `mul!(y, A, x)` vs. `A_mul_B!(y, LinearMap(A), x)`, and there is no difference. I'll remove the macros then.